.github/workflows/deploy_from_develop.yml

```diff
@@ -26,5 +32,33 @@ jobs:
           git pull origin develop
           su - assistant-bible
           cd deployment
+          docker tag deployment-chatbot:latest deployment-chatbot:backup
+          docker compose down
           docker compose --env-file .env up --force-recreate --build -d
 
+    - name: Test Application Endpoint
+      if: steps.deployment.outcome == 'success'
+      run: |
+        for i in {1..5}; do
+          curl --fail http://{${{ env.HOST }}} && exit 0
```

Comment:
What if a nginx home page got loaded instead of our ui? Would that be captured here?

Reply:
It is checking a if a 200ish response is getting back from the url. It will fix last week issues where there was a 500 error being returned.

Not sure what "nginx home page" failure mode is. Can you elaborate?

In principle the curl command should be replaced with a "pytest test/integration" where we run a full sweet of integration tests.

Reply:
I am wondering would we get a 200sh or 500sh status code when this page loads up
https://www.nginx.com/blog/welcome-to-nginx-on-my-favourite-website/
May be it is the same thing you are already handling here

Reply:
So the failure mode is that the nginx server is running and listening but the underlaying uvicorn server is not.
In that case nginx will return a 200 so that case will not be cover here.  It would need a more meaningful test suite for that.